gridded_etl_tools/utils/zarr_methods.py

```diff
@@ -1209,3 +1212,172 @@ def are_times_in_expected_order(self, times: tuple[datetime.datetime], expected_
+        """
+        if self.skip_post_parse_qc:
+            self.info("Skipping post-parse quality check since the source data is different from the parsed data")
+        else:
+            self.info("Beginning post-parse quality check of the parsed dataset")
+            start_checking = time.perf_counter()
+            # Instantiate needed objects
+            self.set_key_dims()  # in case running w/out Transform/Parse
+            prod_ds = self.get_prod_update_ds()
+            # Run the data check N times, incrementing after every successfuly check
+            i = 0
+            while i <= checks:
+                random_coords = self.get_random_coords(prod_ds)
+                orig_ds, _ = self.get_original_ds()
+                i += self.check_value(random_coords, orig_ds, prod_ds, threshold)
+                # Theoretically this could loop endlessly if all input files don't match the prod dataset
+                # in the time dimension. While improbable, let's build an automated exit just in case
+                if time.perf_counter() - start_checking > 1200:
+                    self.info(
+                        f"Breaking from checking loop after \
+                              {datetime.timedelta(seconds=time.perf_counter() - start_checking)} \
+                              to prevent infinite checks"
+                    )
+                    break
+
+            self.info(
+                f"Values check successfully passed.\
+                      Checking dataset took {datetime.timedelta(seconds=time.perf_counter() - start_checking)}"
+            )
+
+        return True
+
+    def get_prod_update_ds(self) -> xr.Dataset:
+        """
+        Get the prod dataset and filter it to the temporal extent of the latest update
+
+        Returns
+        -------
+        prod_ds
+            The production dataset filtered to only the temporal extent of the latest update
+        """
+        prod_ds = self.store.dataset()
+        update_date_range = slice(
+            datetime.datetime.strptime(prod_ds.attrs["update_date_range"][0], "%Y%m%d%H"),
+            datetime.datetime.strptime(prod_ds.attrs["update_date_range"][1], "%Y%m%d%H"),
+        )
+        time_select = {self.time_dim: update_date_range}
+        return prod_ds.sel(**time_select)
+
+    def get_original_ds(self) -> tuple[xr.Dataset, pathlib.Path]:
+        """
+        Get the original dataset and format it equivalently to the production dataset
+
+        Returns
+        ----------
+        orig_ds
+            The original dataset, unformatted
+        orig_file_path
+            The pathlib.Path to the randomly selected original file
+        """
+        orig_file_path = random.choice(list(self.input_files()))
+        if self.protocol == "file":
+            raw_ds = xr.open_dataset(orig_file_path)
+        # Presumes that use_local_zarr_jsons is enabled. This avoids repeating the DL from S#
```

Comment:
We don't necessarily have to assume, if it makes sense we could just skip or throw an exception depending on the importance if use_local_zarr_jsons is not enabled

Reply:
I’m into this